### DISCUSSION TITLE
Getting Scheduled Tasks to Work with Live Reloading

### DISCUSSION THREAD

**Discussion by jack-farley** [quarkusio/quarkus#32247] [Q&A] [answered]
I'm working on an application that is accessing an external API and needs to store an access token for that API. To do that, I have a scheduled event every 12 hours that grabs a new access token and stores it in memory. This generally works great, since the scheduled task runs on start-up and then continuously after that to prevent the token from expiring. 

However, when I use quarkus in dev mode and make a change, live reloading seems to get triggered by the first API call that I make to the application, and this API call is handled before the scheduled event is rerun. Thus, whenever the application is live reloaded, the first API call fails and then I immediately see the scheduled event run and fetch a new access token. 

Any ideas on how to fix this?

[a comment by an automated account is not shown]

**Reply by maxandersen** [reply]
Hmm. So you have a need that whatever the scheduled event is doing is executed before the first api is done ?

Maybe @mkouba can think of a smarter way but if your code has an explicit dependency on something being setup/executed maybe have a check or even interceptor that checks the token has been refreshed and if not does it without relying on the implicit scheduled event? 

Even without hotreload this is a good principle to follow anyway as it makes your app more resilient.

**Reply by mkouba** [reply]
Max is right. I suppose that the token is cached e.g. in a `@Singleton` bean called `TokenHolder`, and accessed via a method like `TokenHolder#getToken()` and the scheduled method updates the token regularly, i.e. calls `TokenHolder#updateToken()`. It would be reasonable to modify the `TokenHolder#getToken()` in a way that if no token is found then fetch it immediately. Alternatively, you can fetch the token [during application startup](https://quarkus.io/guides/cdi-reference#startup_event).

**Reply by jack-farley** [reply]
Thanks for the suggestions. I went with the application startup approach, and that seemed to work well.
